lib/internal/http2/core.js

```diff
@@ -1912,10 +1912,19 @@ class Http2Stream extends Duplex {
 
     let req;
 
+    // writeGeneric does not destroy on error and we cannot enable autoDestroy,
+    // so make sure to destroy on error.
+    const callback = (err) => {
+      if (err) {
+        this.destroy(err);
+      }
+      cb(err);
+    };
+
     if (writev)
-      req = writevGeneric(this, data, cb);
+      req = writevGeneric(this, data, callback);
     else
-      req = writeGeneric(this, data, encoding, cb);
+      req = writeGeneric(this, data, encoding, callback);
```

Comment:
I feel this might need some tests.

Reply:
@addaleax: How do I from a test trigger an error from writeGeneric?